src/solvers/smt/smt_conv.cpp

```diff
@@ -675,7 +675,17 @@ smt_astt smt_convt::convert_ast(const expr2tc &expr)
   }
   case expr2t::typecast_id:
   {
-    a = convert_typecast(expr);
+    const typecast2t &cast = to_typecast2t(expr);
+    if(
+      int_encoding && is_floatbv_type(cast.from->type) &&
+      is_floatbv_type(cast.type))
```

Comment:
Do you think it would make sense to move this code path to `convert_typecast()` as suggested earlier in order to keep the cases in one place?
>  My guess would be that the if (is_floatbv_type(cast.type) && int_encoding) case is missing in smt_convt::convert_typecast(), because it is handled for fixedbv there.

Reply:
I concur. 
Done, moved the changes to `convert_typecast()`.